packages/router-core/src/router.ts

```diff
@@ -2098,6 +2099,28 @@ export class RouterCore<
   }
 
   load: LoadFn = async (opts?: { sync?: boolean }): Promise<void> => {
+    // If this router has a basepath, only respond to paths within scope.
```

Comment:
if we add support for opting out of navigation, we need to check where that should happen best.
if that's here, then i would rather add an optional callback function that can execute arbitrary user code instead of adding this logic here.